rednotebook/data.py

```diff
@@ -200,41 +200,54 @@ def search(self, text, tags):
                         add_text_to_results = True
             if add_text_to_results:
                 results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
-        elif text in str(self):
-            # Date contains searched text.
-            results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
         else:
-            text_result = self.search_in_text(text)
+            for word in words:
+                # Any of the word matches with the date.
+                if word in str(self):
+                    # We don't want to show search results from the text matching
+                    # this date.
+                    words.remove(word)
+                    results.append(get_text_with_dots(self.text, 0, TEXT_RESULT_LENGTH))
+            # if all the words matched agains the date, return
+            if not words:
+                return str(self), words
+            text_result = self.search_in_text(words)
             if text_result:
                 results.append(text_result)
-            results.extend(self.search_in_categories(text))
+            results.extend(self.search_in_categories(words))
         return str(self), results
 
-    def search_in_text(self, search_text):
-        occurence = self.text.upper().find(search_text.upper())
-
-        # Check if search_text is in text
-        if occurence < 0:
-            return None
-
-        found_text = self.text[occurence : occurence + len(search_text)]
-        result_text = get_text_with_dots(
-            self.text, occurence, occurence + len(search_text), found_text
+    def search_in_text(self, words):
+        """All words should be present in the text"""
```

Comment:
`If all words are in the text, return a suitable text substring. Otherwise, return None.`